independent-projects/junit5-virtual-threads/src/main/java/io/quarkus/test/junit5/virtual/ShouldNotPin.java

```diff
@@ -8,9 +8,10 @@
 /**
  * Marker indicating that the test method or class should not pin the carrier thread.
  * If, during the execution of the test, a virtual thread pins the carrier thread, the test fails.
+ * However occasional pin can still be allowed by setting {@code atMost} value
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ ElementType.METHOD, ElementType.TYPE })
 public @interface ShouldNotPin {
-
+    int atMost() default 0;
```

Comment:
Might be useful to repeat part of the javadoc you added above.

Reply:
Added java doc. Thanks!